lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
Can you tell me more? How different are these configurations? I guess the only reason to create a new compiler class is if you believe that the xl and xl_r could somehow be incompatible, or if you think that a user would realistically like to keep xl and xl_r builds separate.

I just checked and the `xl.py` compiler class already has things like `xlc_r` as possible compiler names, so we already support this. If we do decide to split them into two compilers, we would have to remove these names from `xl.py`. Otherwise I don't know which would be detected.